lib/Connector/CapConnector.php

```diff
@@ -0,0 +1,487 @@
+            $area['radius'] = $radius;
+            return $area;
+        }
+
+        // Check for a polygon area element
+        $polygon = $this->getAreaData('polygon');
+        if ($polygon) {
+            // Split the polygon data into multiple points ("lat1,lon1 lat2,lon2 ...")
+            $points = explode(' ', $polygon);
+
+            // Array to store multiple coordinates
+            $polygonPoints = [];
+
+            foreach ($points as $point) {
+                $coords = explode(',', $point);
+                $polygonPoints[] = ['lat' => $coords[0], 'lon' => $coords[1]];
+            }
+
+            $area['type'] = 'polygon';
+            $area['points'] = $polygonPoints;
+        }
+
+        return $area;
+    }
+
+    /**
+     * Checks if the provided display coordinates are inside a defined area (circle or polygon).
+     * If no area coordinates are available, it returns false.
+     *
+     * @param float $displayLatitude
+     * @param float $displayLongitude
+     * @param array $areaCoordinates The coordinates defining the area (circle or polygon).
+     *
+     * @return bool
+     */
+    private function isWithinArea(float $displayLatitude, float $displayLongitude, array $areaCoordinates): bool
+    {
+        if (empty($areaCoordinates)) {
+            // No area coordinates available
+            return false;
+        }
+
+        // Initialize the display coordinate
+        $displayCoordinate = new Coordinate($displayLatitude, $displayLongitude);
+
+        if ($areaCoordinates['type'] == 'circle') {
+            // Initialize the circle's coordinate and radius
+            $centerCoordinate = new Coordinate($areaCoordinates['center']['lat'], $areaCoordinates['center']['lon']);
+            $radius = $areaCoordinates['radius'];
+
+            // Check if the display is within the specified radius of the center coordinate
+            if ($centerCoordinate->hasSameLocation($displayCoordinate, $radius)) {
+                return true;
+            }
+        } else {
+            // Initialize a new polygon
+            $geofence = new Polygon();
+
+            // Add each point to the polygon
+            foreach ($areaCoordinates['points'] as $point) {
+                $geofence->addPoint(new Coordinate($point['lat'], $point['lon']));
+            }
+
+            // Check if the display is within the polygon
+            if ($geofence->contains($displayCoordinate)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param ScheduleCriteriaRequestInterface $event
+     * @return void
+     * @throws ConfigurationException
+     */
+    public function onScheduleCriteriaRequest(ScheduleCriteriaRequestInterface $event): void
+    {
+        // Initialize Emergency Alerts schedule criteria parameters
+        $event->addType('emergency_alerts', __('Emergency Alerts'))
+                ->addMetric('status', __('Status'))
```

Comment:
Should we make these more specific to avoid conflicts with other criteria metrics?